backend/app/models.py

```diff
@@ -159,17 +118,44 @@ def __init__(self, citizen_list=None, country="South Africa"):
         self.country = country
         self.population_size = 0
 
-    def create_citizens(self, number_to_create):
+    # pylint: disable=too-many-nested-blocks
```

Comment:
This really _is_ too many nested blocks -- rather than disable this linter check, let's fix it. 

The main problem is that this is two completely separate functions disguised as one. If you find that you have an if/else on the presence of a single argument and there's nothing else going on the function, that's probably a sign you should refactor. What you want here is a new function for the version that uses `demographic_data`, and probably leave the old `create_citizens` as is (except rename it, since you'll have two separate funcs... maybe `create_citizens_budget_sim` and `create_citizens_campaign_game`

Reply:
Please also add a backend test for this new function